src/utilities/tleHandling.py

```diff
@@ -0,0 +1,449 @@
+    meanAnomaly_str = line2[43:51] # Mean Anomaly in degrees
+    meanMotion_str = line2[52:63] # Mean Motion in revolutions per day
+    revNo_str = line2[63:68] # Revolution number at epoch
+    checkSumLine2 = int(line2[68:69])
+    calcChecksum2 = _calcTleChecksum([line2_str, satID_2_str, inclDeg_str, OmegaDeg_str, e_str, omega_str, meanAnomaly_str, meanMotion_str, revNo_str])
+    if checkSumLine2 != calcChecksum2:
+        raise ValueError(f"ERROR: _parseTle() checksum mismatch for line2. The calculated checksum is {calcChecksum2} while the provided checksum is {checkSumLine2}.")
+
+    # Parsing numerical values getting
+    satName = line0.strip()
+    noradId = int(satId1_str)
+    classification = {'U': 'Unclassified', 'C': 'Classified', 'S': 'Secret'}.get(class_str, 'Unknown')
+
+    epochYear = _parseTleYear(line1[18:20])
+    tle_datetime = dt.datetime(epochYear, 1, 1) + dt.timedelta(days=float(epochDay_str) - 1) # subtract one day because Jan 1 is day 1 (not day 0)
+    tle_datetime = tle_datetime.replace(tzinfo=dt.timezone.utc)
+
+    # Converting TLE strings to physical values
+    nDot = _firstDeriv2ballistCoef(nDot_str) # dn/dt in [rev/day^2]
+    nDotDot = _tleStr2physVal(nDotDot_str) # d^2n/dt^2 in [rev/day^3]
+    bStar = _tleStr2physVal(bStar_str)
+    propagator = {'1': 'SGP4', '2': 'SDP4', '3': 'SGP8', '4': 'SDP8', '0': 'Unknown'}.get(ephemType_str, 'Unknown')
+    elemSetNo = int(elemSetNo_str)
+
+    # Converting TLE information to classical orbital elements and write to TleData class
+    oe = om.ClassicElements()
+    oe.i = np.deg2rad(float(inclDeg_str.strip()))
+    oe.Omega = np.deg2rad(float(OmegaDeg_str.strip()))
+    oe.e = float(("0." + e_str).strip())
+    oe.omega = np.deg2rad(float(omega_str.strip()))
+    revDay = float(meanMotion_str.strip())
+    oe.a = (om.MU_EARTH / ((revDay * 2.0 * np.pi) / SEC_PER_DAY) ** 2) ** (1.0 / 3.0) * 1e3
+    M = np.deg2rad(float(meanAnomaly_str.strip()))
+    oe.f = om.E2f(om.M2E(M, oe.e), oe.e)
+
+    tleData = TleData(oe=oe, tleEpoch=tle_datetime)
+    # Optional data
+    tleData.satName = satName
+    tleData.noradID = noradId
+    tleData.classification = classification
+    tleData.revAtEpoch = float(revNo_str.strip())
+    tleData.propagator = propagator
+    tleData.elemSetNo = elemSetNo
+    tleData.nDot = nDot
+    tleData.nDotDot = nDotDot
+    tleData.bStar = bStar
+    tleData.launchDate = dt.datetime(_parseTleYear(launchYear_str), 1, 1)
+    tleData.launchNo = int(launchNo_str)
+    tleData.pol = pol_str
+
+    return tleData
+
+def satTle2elem(tle_path: str):
+    """
+    Convert the TLEs of a constellation to classical orbital elements for each satellite.
+
+    :param tle_path: path to a TLE with one or multiple satellites in either 2LE or 3LE format
+    :return: elementsList: list of classical orbital elements for each satellite
+             metadataList: list of dictionaries containing TLE metadata for each satellite
+    """
```

Comment:
Nitpick:

This method is doing a lot of things. Generally once a function hits around 100 lines I start thinking on if I can break it up more. This is more of a personal pref. thing though so I will defer to @schaubh.